common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -123,7 +125,10 @@ public void connectionTerminated(Channel channel) {
     for (Map.Entry<Long, StreamState> entry: streams.entrySet()) {
       StreamState state = entry.getValue();
       if (state.associatedChannel == channel) {
-        streams.remove(entry.getKey());
+        synchronized (lock) {
+          streams.remove(entry.getKey());
```

Comment:
Is locking on `streams` not sufficient?

Reply:
> Is locking on `streams` not sufficient?
 
Haven't thought of a particularly good solution to consistency and performance conflicts

Reply:
I just mean you could use it as the mutex but maybe that has undesirable interactions with how it synchronizes other access. OK no big deal.

Reply:
> I just mean you could use it as the mutex but maybe that has undesirable interactions with how it synchronizes other access. OK no big deal.

Yea